apps/daimo-mobile/src/view/screen/LandlineBankTransfer.tsx

```diff
@@ -0,0 +1,263 @@
+        params,
+      });
+    if (money != null) goTo({ landlineAccount });
+    else if (nav.canGoBack()) nav.goBack();
+    else goHome();
+  }, [nav, money, landlineAccount]);
+
+  const sendDisplay = (() => {
+    if (money == null)
+      return (
+        <SendChooseAmount
+          landlineAccount={landlineAccount}
+          onCancel={goBack}
+          daimoChain={daimoChain}
+        />
+      );
+    else return <SendConfirm {...{ account, landlineAccount, memo, money }} />;
+  })();
+
+  return (
+    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
+      <View style={ss.container.screen}>
+        <ScreenHeader
+          title={landlineAccount.bankName}
+          onBack={goBack}
+          onExit={goHome}
+        />
+        <Spacer h={8} />
+        {sendDisplay}
+      </View>
+    </TouchableWithoutFeedback>
+  );
+}
+
+function landlineAccountToContact(
+  landlineAccount: LandlineAccount
+): BridgeBankAccountContact {
+  return {
+    type: "bridgeBankAccount",
+    addr: landlineAccount.liquidationAddress,
+    bankName: landlineAccount.bankName,
+    lastFour: landlineAccount.lastFour,
+    bankLogo: landlineAccount.bankLogo,
+  };
+}
+
+function SendChooseAmount({
+  landlineAccount,
+  daimoChain,
+  onCancel,
+}: {
+  landlineAccount: LandlineAccount;
+  daimoChain: DaimoChain;
+  onCancel: () => void;
+}) {
+  const recipient = landlineAccountToContact(landlineAccount);
+
+  // Select how much
+  const [money, setMoney] = useState(zeroUSDEntry);
+
+  // Select what for
+  const [memo, setMemo] = useState<string | undefined>(undefined);
+
+  // Once done, update nav
+  const nav = useNav();
+  const setSendAmount = () =>
+    nav.navigate("DepositTab", {
+      screen: "LandlineTransfer",
+      params: { money, memo, landlineAccount },
+    });
+
+  // Validate memo
+  const rpcHook = env(daimoChain).rpcHook;
+  const result = rpcHook.validateMemo.useQuery({ memo });
+  const memoStatus = result.data;
+
+  return (
+    <View>
+      <Spacer h={24} />
+      <ContactDisplay contact={recipient} />
+      <Spacer h={24} />
+      <AmountChooser
+        moneyEntry={money}
+        onSetEntry={setMoney}
+        showAmountAvailable
+        autoFocus
+      />
+      <Spacer h={16} />
+      <SendMemoButton
+        memo={memo}
+        memoStatus={memoStatus}
+        setMemo={setMemo}
+        daimoChain={daimoChain}
+      />
+      <Spacer h={16} />
+      <View style={styles.buttonRow}>
+        <View style={styles.buttonGrow}>
+          <ButtonBig type="subtle" title="CANCEL" onPress={onCancel} />
+        </View>
+        <View style={styles.buttonGrow}>
+          <ButtonBig
+            type="primary"
+            title="CONFIRM"
+            onPress={setSendAmount}
+            disabled={
+              money.dollars === 0 || (memoStatus && memoStatus !== "ok")
+            }
+          />
+        </View>
+      </View>
+      <Spacer h={14} />
+      <PublicWarning />
```

Comment:
how big is the diff between this `SendChooseAmount` / `SendConfirm` and the one in SendTransferScreen ?

Reply:
main difference is the `nav`. in SendTransferScreen, the nav goes back to the transfer screen, whereas we want it to go back to the deposit screen in this component

also, we might want to add a component for choosing between deposit and withdraw in the landline `SendChooseAmount`, so it might be ugly to have a bunch of conditionals if we were to combine the components

Reply:
fine to ship as-is

i have ideas for follow-up. we can sketch options tomorrow